nvme-print.c

```diff
@@ -893,61 +918,121 @@ void nvme_show_lba_status_info(__u32 result)
 const char *nvme_host_metadata_type_to_string(enum nvme_features_id fid,
 					      __u8 type)
 {
-       switch (fid) {
-       case NVME_FEAT_FID_ENH_CTRL_METADATA:
-       case NVME_FEAT_FID_CTRL_METADATA:
-	       switch (type) {
-	       case NVME_CTRL_METADATA_OS_CTRL_NAME:
-		       return "Operating System Controller Name";
-	       case NVME_CTRL_METADATA_OS_DRIVER_NAME:
-		       return "Operating System Driver Name";
-	       case NVME_CTRL_METADATA_OS_DRIVER_VER:
-		       return "Operating System Driver Version";
-	       case NVME_CTRL_METADATA_PRE_BOOT_CTRL_NAME:
-		       return "Pre-boot Controller Name";
-	       case NVME_CTRL_METADATA_PRE_BOOT_DRIVER_NAME:
-		       return "Pre-boot Driver Name";
-	       case NVME_CTRL_METADATA_PRE_BOOT_DRIVER_VER:
-		       return "Pre-boot Driver Version";
-	       case NVME_CTRL_METADATA_SYS_PROC_MODEL:
-		       return "System Processor Model";
-	       case NVME_CTRL_METADATA_CHIPSET_DRV_NAME:
-		       return "Chipset Driver Name";
-	       case NVME_CTRL_METADATA_CHIPSET_DRV_VERSION:
-		       return "Chipset Driver Version";
-	       case NVME_CTRL_METADATA_OS_NAME_AND_BUILD:
-		       return "Operating System Name and Build";
-	       case NVME_CTRL_METADATA_SYS_PROD_NAME:
-		       return "System Product Name";
-	       case NVME_CTRL_METADATA_FIRMWARE_VERSION:
-		       return "Firmware Version";
-	       case NVME_CTRL_METADATA_OS_DRIVER_FILENAME:
-		       return "Operating System Driver Filename";
-	       case NVME_CTRL_METADATA_DISPLAY_DRV_NAME:
-		       return "Display Driver Name";
-	       case NVME_CTRL_METADATA_DISPLAY_DRV_VERSION:
-		       return "Display Driver Version";
-	       case NVME_CTRL_METADATA_HOST_DET_FAIL_REC:
-		       return "Host-Determined Failure Record";
-	       default:
-		       return "Unknown Controller Type";
-	       }
-       case NVME_FEAT_FID_NS_METADATA:
-	       switch (type) {
-	       case NVME_NS_METADATA_OS_NS_NAME:
-		       return "Operating System Namespace Name";
-	       case NVME_NS_METADATA_PRE_BOOT_NS_NAME:
-		       return "Pre-boot Namespace Name";
-	       case NVME_NS_METADATA_OS_NS_QUAL_1:
-		       return "Operating System Namespace Name Qualifier 1";
-	       case NVME_NS_METADATA_OS_NS_QUAL_2:
-		       return "Operating System Namespace Name Qualifier 2";
-	       default:
-		       return "Unknown Namespace Type";
-	       }
-       default:
-	       return "Unknown Feature";
-       }
+	switch (fid) {
```

Comment:
please do not do whitespace change when adding new code. split this into a separate patch if it is actually necessary.